src/pudl_archiver/zenodo/api_client.py

```diff
@@ -80,7 +81,14 @@ def __init__(
         self.deposition: Deposition = None
         self.new_deposition: Deposition = None
         self.deposition_files: dict[str, DepositionFile] = {}
-        self.changed = False
+
+        self.dry_run = dry_run
+        # We accumulate changes in a changeset, then apply - makes dry runs and testing easier.
+        # upload takes (source: IOBase | Path, dest: str) tuples; delete just takes the str key to delete.
+        self.changeset: dict[str, tuple[io.IOBase | Path, str] | str] = {
```

Comment:
Type annotation here is a little confusing at first, could maybe use a class or alias here. I could also see this being overtaken in a future round of refactoring (like the logic refactor you mentioned), so I don't think it's too big of a deal

Reply:
Yeah, I thought about making, like, an `UploadSpec` namedtuple or something - not a lot of overhead, I'll make it happen.